pymathics/natlang/normalization.py

```diff
@@ -0,0 +1,301 @@
+from mathics.core.evaluation import Evaluation
+from mathics.core.list import ListExpression
+
+from pymathics.natlang.spacy import _cases, _pos_tags, _position, _SpacyBuiltin
+
+
+class DeleteStopwords(_SpacyBuiltin):
+    """
+    Delete <url>:stop words:https://en.wikipedia.org/wiki/Stop_word</url>(\
+    <url>:WMA:
+    https://reference.wolfram.com/language/ref/DeleteStopwords.html</url>\
+    )
+
+    <dl>
+      <dt>'DeleteStopwords[$list$]'
+      <dd>returns the words in $list$ without stopwords.
+
+      <dt>'DeleteStopwords[$string$]'
+      <dd>returns $string$ without stopwords.
+    </dl>
+
+    ## This has changed since old versions of natlang, and I am
+    ## not sure the old behavior was correct.
+    >> DeleteStopwords[{"Somewhere", "over", "the", "rainbow"}]
+     = ...
+    ## = {rainbow}
+
+    >> DeleteStopwords["There was an Old Man of Apulia, whose conduct was very peculiar"]
+     = Old Man Apulia, conduct peculiar
+    """
+
+    summary_text = "Remove stopwords from a text"
+
+    def eval_list(self, li, evaluation: Evaluation, options: dict) -> ListExpression:
+        "DeleteStopwords[li_List, OptionsPattern[DeleteStopwords]]"
+        is_stop = self._is_stop_lambda(evaluation, options)
+
+        def filter_words(words):
+            for w in words:
+                s = w.get_string_value()
+                if s is not None:
+                    yield String(s)
+                elif is_stop is not None and is_stop(s) is not None:
+                    yield String(s)
+
+        return ListExpression(*list(filter_words(li.elements)))
+
+    def eval_string(self, s: String, evaluation: Evaluation, options: dict):
+        "DeleteStopwords[s_String, OptionsPattern[DeleteStopwords]]"
+        doc = self._nlp(s.value, evaluation, options)
+        if doc:
+            is_stop = self._is_stop_lambda(evaluation, options)
+            if is_stop:
+
+                def tokens():
+                    for token in doc:
+                        if not is_stop(token.text):
+                            yield token.text_with_ws
+                        else:
+                            yield token.whitespace_.strip()
+
+                return String("".join(tokens()))
+
+
+class TextCases(_SpacyBuiltin):
+    """
+    <url>:WMA:
+    https://reference.wolfram.com/language/ref/TextCases.html</url>
+
+    <dl>
+      <dt>'TextCases[$text$, $form$]'
+      <dd>returns all elements of type $form$ in $text$ in order of their appearance.
+    </dl>
+
+    >> TextCases["I was in London last year.", "Pronoun"]
+     = {I}
+
+    >> TextCases["I was in London last year.", "City"]
+     = {London}
+
+    ## >> TextCases[Import["ExampleData/EinsteinSzilLetter.txt"], "Person", 3][[2;;3]]
+    ##  = {L. Szilard, Joliot}
+
+    >> TextCases["Anne, Peter and Mr Johnes say hello.", "Person", 3][[2;;3]]
+     = {Peter, Johnes}
+
+    """
+
+    summary_text = "List the cases of words of a certain form in a text"
```

Comment:
Again we have sometimes "List" and sometimes "list". Use lowercase here and below.